packages/playwright-ct-angular/registerSource.mjs

```diff
@@ -0,0 +1,137 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// @ts-check
+// This file is injected into the registry as text, no dependencies are allowed.
+
+/** @typedef {import('../playwright-ct-core/types/component').ObjectComponent} ObjectComponent */
+
+import 'zone.js';
+import { reflectComponentType } from '@angular/core';
+import { getTestBed, TestBed } from '@angular/core/testing';
+import {
+  BrowserDynamicTestingModule,
+  platformBrowserDynamicTesting,
+} from '@angular/platform-browser-dynamic/testing';
+
+/** @type {WeakMap<import('@angular/core/testing').ComponentFixture, Record<string, import('rxjs').Subscription>>} */
+const __pwOutputSubscriptionRegistry = new WeakMap();
+
+/** @type {Map<string, import('@angular/core/testing').ComponentFixture>} */
+const __pwFixtureRegistry = new Map();
+
+getTestBed().initTestEnvironment(
+    BrowserDynamicTestingModule,
+    platformBrowserDynamicTesting(),
+);
+
+/**
+ * @param {ObjectComponent} component
+ */
+async function __pwRenderComponent(component) {
+  const componentMetadata = reflectComponentType(component.type);
+  if (!componentMetadata?.isStandalone)
+    throw new Error('Only standalone components are supported');
+
+  TestBed.configureTestingModule({
+    imports: [component.type],
+  });
+
+  await TestBed.compileComponents();
+
+  const fixture = TestBed.createComponent(component.type);
+  fixture.nativeElement.id = 'root';
+
+  __pwUpdateProps(fixture, component.props);
+  __pwUpdateEvents(fixture, component.on);
+
+  fixture.autoDetectChanges();
+
+  return fixture;
+}
+
+/**
+ * @param {import('@angular/core/testing').ComponentFixture} fixture
+ */
+function __pwUpdateProps(fixture, props = {}) {
+  for (const [name, value] of Object.entries(props))
+    fixture.componentRef.setInput(name, value);
+}
+
+/**
+ * @param {import('@angular/core/testing').ComponentFixture} fixture
+ */
+function __pwUpdateEvents(fixture, events = {}) {
+  const outputSubscriptionRecord =
+    __pwOutputSubscriptionRegistry.get(fixture) ?? {};
+  for (const [name, listener] of Object.entries(events)) {
+    /* Unsubscribe previous listener. */
+    outputSubscriptionRecord[name]?.unsubscribe();
+
+    const subscription = fixture.componentInstance[
+        name
+    ].subscribe((/** @type {unknown} */ event) => listener(event));
+
+    /* Store new subscription. */
+    outputSubscriptionRecord[name] = subscription;
+  }
+
+  /* Update output subscription registry. */
+  __pwOutputSubscriptionRegistry.set(fixture, outputSubscriptionRecord);
+}
+
+window.playwrightMount = async (component, rootElement, hooksConfig) => {
```

Comment:
I'm coming here from https://github.com/sand4rt/playwright/pull/5#discussion_r1602736402, which discussed the need to render a HTML template.

After giving it some more thought, I'd like to change my opinion.
With standalone components I don't see a real need support HTML templates.
Mainly because this will only(?) be used for Angular Component Tests, other TL libraries also don't support this AFAIK.

Before standalone components testing a directive contained more setup, but with the current state it's as simple as with other libraries. Yes, we'd have to create a container component manually, but this will also give more control to the user and less dependencies between Playwright and Angular. In my opinion this outweighs the benefits a user gets to create an inline template.

Without support:

```ts
test('renders a directive', async () => {
  // can also be declared outside of the scope to be reusable
  @Component({
    template: `<div appSpoiler data-testid="dir"></div>`,
    imports: [SpoilerDirective],
    standalone: true,
  })
  class WrapperComponent {}

  await mount(WrapperComponent );
});
``` 

With support:

```ts
test('renders a directive', async () => {
  await mount( `<div appSpoiler data-testid="dir"></div>`, {
     /* I suppose some configuration needs to come here to import the Directive */
  });
});
```

Reply:
This does align with Playwright's suggestion for writing [test stories](https://playwright.dev/docs/test-components#test-stories). However the first example needs to be extracted to a separate file since it's not importable. It would be nice to have [something like storybook](https://gist.github.com/chronospatian/88e8433382e6e23d1fbb40da08afe055) to reduce the boilerplate. Maybe a future version could consume storybook CSF directly?

Reply:
Thanks for the comments! @timdeschryver Yeah, that's what i'm leaning towards as well. I'll tag you/this when the discussion pops-up again after the merge.

@chronospatian The Storybook team is working on this:) https://github.com/microsoft/playwright/pull/26592#issuecomment-1890927499

Reply:
Some more info on this: https://storybook.js.org/blog/portable-stories-for-playwright-ct 

related to: https://github.com/microsoft/playwright/issues/30981

Reply:
There are different tradeoffs to consider.
For most cases, a test container / story component / portable storybook story is enough.

That said, there are cases where each test (or subgroup of tests) might need a totally different template construction. This can cause some clutter with a different component for each test.
Mounting template in the tests can make some tests a bit more readable with less clutter.
On the other hand, mounting templates will necessarily require JIT.
 
IMHO, this should be up to the testing strategy of the users. 
Also, this could be handled by a 3rd party library like `@jscutlery/playwright-ct-angular` which we are planning to maintain as an incubator for the official package ideas and also as a wrapper of the official support.